### ISSUE TITLE
Rule Request: Recommended SwiftLint version

### ISSUE THREAD

**Issue by mflint** [realm/SwiftLint#3567]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

When working in a team, we inevitably have people running different versions of SwiftLint - so developers often see different output from SwiftLint. And if the CI machine has a different version of SwiftLint, builds can unexpectedly fail.

Would you consider adding a recommended version to the `.swiftLint.yml` file?

### How might this work?

Two new optional values in the `.swiftlint.yml` file:

```yaml
recommended_version: 0.44.0
```

SwiftLint would emit a warning if the running version is not _exactly_ 0.44.0.

```yaml
minimum_recommended_version: 0.44.0
```

SwiftLint would emit a warning if the version is not 0.44.0 or later.

If neither `recommended_version` nor `minimum_recommended_version` are specified in the configuration file, then no version check would be performed.

**Comment by mflint** [comment]
Ack - duplicate of #2592. Closing.
